host-controller/src/main/java/org/jboss/as/host/controller/discovery/S3Discovery.java

```diff
@@ -130,7 +130,7 @@ public int compare(DomainControllerData data, DomainControllerData otherData) {
                     }
                     return -1;
                 }});
-            writeToFile(data, MASTER);
+            writeToFile(data, PRIMARY);
```

Comment:
These changes are out of the scope of this Jira. Changing [S3Discovery.java](https://github.com/wildfly/wildfly-core/pull/5031/files#diff-d64c6ab5368f0778182f6700fb69dc7e11c784f1825f1f805f96d019d2c912c4) due to the attribute name sound seems unrelated. Correct me if I am wrong, otherwise, we should not add changes out of the Jira scope. That will help to organize the work to do and avoid conflicts.